pkg/providers/amifamily/ami.go

```
@@ -133,6 +135,17 @@ func (p *DefaultProvider) DescribeImageQueries(ctx context.Context, nodeClass *v
 	return queries, nil
 }
 
+func (p *DefaultProvider) updateFilterFromCustomParameter(ctx context.Context, ssmParameterName string, idFilter *ec2types.Filter) {
+	imageID, err := p.ssmProvider.GetCustomParameter(ctx, ssm.Parameter{
+		Name: ssmParameterName,
+	})
+	if err != nil {
+		log.FromContext(ctx).WithValues("ssmParameterName", ssmParameterName).V(1).Error(err, "parameter not found")
```

We need to account for more error scenarios here than just "not found". It's possible we weren't able to resolve the SSM parameter due to transient networking issues, or due to insufficient IAM permissions. If it is not found, this behavior seems correct - we should just not add anything to the filter and carry on. But if it's some other form of error, we would want to bubble up the error. We already have a [function for identifying NotFound errors from EC2](https://github.com/aws/karpenter-provider-aws/blob/0a8251d6eba027c7d34acaf9d0fee2fc40832b67/pkg/errors/errors.go#L62-L73), we would just need to add the code for SSM.

I imagine we'd probably want something along these lines:
```go
id, err := p.ssmProvider.GetParameter(...)
if err != nil {
    if !errors.IsNotFound(err) {
        return "", fmt.Errorf("resolving ssm parameter, %w", err)
    }
    log.FromContext(ctx).WithValues("parameter", ssmParameterName).V(1).Info("failed to discover ami for ssm parameter, parameter not found")
    return "", nil
}
return id, nil

```